aeron-driver/src/main/java/io/aeron/driver/PublicationParams.java

```diff
@@ -488,6 +490,28 @@ private void getUntetheredWindowLimitTimeout(final ChannelUri channelUri, final
             channelUri, UNTETHERED_WINDOW_LIMIT_TIMEOUT_PARAM_NAME, ctx.untetheredWindowLimitTimeoutNs());
     }
 
+    private void getUntetheredLingerTimeout(final ChannelUri channelUri, final MediaDriver.Context ctx)
+    {
+        final String windowLimitTimeoutString = channelUri.get(UNTETHERED_WINDOW_LIMIT_TIMEOUT_PARAM_NAME);
+        final String timeoutString = channelUri.get(UNTETHERED_LINGER_TIMEOUT_PARAM_NAME);
+        if (null != timeoutString)
+        {
+            // Linger timeout was explicitly set for the channel.  Use the set value.
+            untetheredLingerTimeoutNs = SystemUtil.parseDuration(UNTETHERED_LINGER_TIMEOUT_PARAM_NAME, timeoutString);
+        }
+        else if (null != windowLimitTimeoutString)
+        {
+            // Window limit timeout was explicitly set for the channel.  Use it.
+            untetheredLingerTimeoutNs = untetheredWindowLimitTimeoutNs;
+        }
+        else
+        {
+            // Neither Window Limit Timeout nor Linger Timeout was set for the channel.
+            // Use the Context value for Linger.
+            untetheredLingerTimeoutNs = ctx.untetheredLingerTimeoutNs();
+        }
+    }
+
```

Comment:
We should only check for `UNTETHERED_WINDOW_LIMIT_TIMEOUT_PARAM_NAME` if `UNTETHERED_LINGER_TIMEOUT_PARAM_NAME` not defined.